Ktisis/Interface/Components/BoneTree.cs

```diff
@@ -32,30 +49,49 @@ public class BoneTree {
 				}
 			}
 		}
+
+		private static bool BoneDoesMatch(Bone bone) => 
+			SearchText != "" && (new [] { bone.UniqueName, bone.LocaleName }).Any(
+				name => name.Contains(SearchText, StringComparison.CurrentCultureIgnoreCase)
+			);
+
```

Comment:
Regarding the 'Criteria' building, you said
> The use of an array only for this, would also have been excessive for its repeated calls, overall performance and impact on garbage collection.

Isn't that also the case here? You can simply replace the array-`Any` with a boolean-OR:

```suggestion
			bone.UniqueName.Contains(SearchText, StringComparison.CurrentCultureIgnoreCase) || bone.LocaleName.Contains(SearchText, StringComparison.CurrentCultureIgnoreCase);
```

(this suggestion assumes the `SearchText != ""` check is moved outside the method)